Source/SvgDocument.cs

```diff
@@ -394,7 +394,10 @@ private static T Open<T>(XmlReader reader) where T : SvgDocument, new()
                     {
                         try
                         {
-                            elemsToStyle = svgDocument.QuerySelectorAll(rule.Selector.ToString(), elementFactory);
+                            var rootNode = new NonSvgElement();
+                            rootNode.Children.Add(svgDocument);
+
+                            elemsToStyle = rootNode.QuerySelectorAll(rule.Selector.ToString(), elementFactory);
                             foreach (var elem in elemsToStyle)

```

Comment:
Ah, what a nice fix! Hadn't thought about this...
There is a test file for this issue, can you add it to the passed tests (provided it passes the test now...)?
Also, can you have a look at #280? This may also be fixed by this one.

Reply:
@mrbean-bremen 

> Ah, what a nice fix! Hadn't thought about this...
There is a test file for this issue, can you add it to the passed tests (provided it passes the test now...)?

Oh... sorry.
I forgot to push.

Reply:
@mrbean-bremen 

> Also, can you have a look at #280? This may also be fixed by this one.

__issue-280-02.svg has improved.
But it seems that there are still other issue.
I will continue to analyse and create another PR.